sentry-compose-helper/src/main/java/io/sentry/compose/gestures/ComposeGestureTargetLocator.java

```diff
@@ -0,0 +1,106 @@
+package io.sentry.compose.gestures;
+
+import androidx.compose.ui.layout.LayoutCoordinatesKt;
+import androidx.compose.ui.layout.ModifierInfo;
+import androidx.compose.ui.node.LayoutNode;
+import androidx.compose.ui.node.Owner;
+import androidx.compose.ui.semantics.SemanticsConfiguration;
+import androidx.compose.ui.semantics.SemanticsModifier;
+import androidx.compose.ui.semantics.SemanticsPropertyKey;
+import io.sentry.internal.gestures.GestureTargetLocator;
+import io.sentry.internal.gestures.UiElement;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@SuppressWarnings("KotlinInternalInJava")
+public final class ComposeGestureTargetLocator implements GestureTargetLocator {
+
+  @Override
+  public @Nullable UiElement locate(
+      @NotNull Object root, float x, float y, UiElement.Type targetType) {
+    @Nullable String targetTag = null;
+
+    if (!(root instanceof Owner)) {
+      return null;
+    }
+
+    final @NotNull Queue<LayoutNode> queue = new LinkedList<>();
+    queue.add(((Owner) root).getRoot());
+
+    while (!queue.isEmpty()) {
+      final @Nullable LayoutNode node = queue.poll();
+      if (node == null) {
+        continue;
+      }
+
+      final boolean isPlaced = node.isPlaced();
+      final boolean inBounds = layoutNodeBoundsContain(node, x, y);
+
+      if (node.isPlaced() && layoutNodeBoundsContain(node, x, y)) {
+        boolean isClickable = false;
+        boolean isScrollable = false;
+        @Nullable String testTag = null;
+
+        final List<ModifierInfo> modifiers = node.getModifierInfo();
+        for (ModifierInfo modifierInfo : modifiers) {
+          if (modifierInfo.getModifier() instanceof SemanticsModifier) {
+            final SemanticsModifier semanticsModifierCore =
+                (SemanticsModifier) modifierInfo.getModifier();
+            final SemanticsConfiguration semanticsConfiguration =
+                semanticsModifierCore.getSemanticsConfiguration();
+            for (Map.Entry<? extends SemanticsPropertyKey<?>, ?> entry : semanticsConfiguration) {
+              final @Nullable String key = entry.getKey().getName();
```

Comment:
I thought we are not allowed to pass nullable values into `switch`? I guess it will throw an NPE if the key is null?